server/auth/github/client_test.go

```diff
@@ -1,3 +1,8 @@
+// Disabled based on https://github.com/remind101/empire/commit/775178497a2ce27a5798a54937dea256c6253dc9
+// "it seems like the mock library cannot detect our functions for
+// any go version over 1.7.6 -- this comment will disable these tests."
+// +build !go1.10
+
```

Comment:
After about an hour of working with this, I think I concur with this. Like you found, the mock library here can't compare instances of http.Request properly and thus can't mock it out properly. 

The best pointer I could find on this front was this article for mocking it out by hand, https://www.thegreatcodeadventure.com/mocking-http-requests-in-golang/, which we could take a look at doing.